docs/work-flow.md

```diff
@@ -0,0 +1,176 @@
+# DC/OS Development Workflow
+
+
+        "... there is absolutely nothing "obvious" about source-control systems, or workflows before you already know them"
+
+Mesosphere DC/OS has two primary repositories.
+
+1. Open Source DC/OS (https://github.com/dcos/dcos)
+2. Closed Source Mesosphere Enterprise DC/OS (https://github.com/mesosphere/dcos-enterprise)
+
+In our design, the closed source Mesosphere Enterprise DC/OS depends on the Open Source DC/OS, [builds on top of the artifact](https://github.com/mesosphere/dcos-enterprise/blob/master/util/build.py) from the Open source DC/OS.
+
+We have designed our DC/OS pull request workflow, in such a way,  we test our Open Source changes
+with the Enterprise consumer,  we keep both versions ready for release at any moment in time.
+It can be cumbersome, and error-prone to deal with two version control systems at the same time.
+A friendly bot called `@mesosphere-mergebot` comes to our rescue and helps the developers with the process.
+
+#### How do I contribute a change to dcos/dcos ?
+
+If a change is only to Enterprise DC/OS, the developer should create the pull request against the Enterprise Repository.
+
+For all contributions, a pull request is against Open Source DC/OS and tested with both Open and
+Enterprise, and we merge the Open Source pull request, and update the reference in Enterprise.
+
+#### What should be my target branch?
+
+All Pull Requests should be against `master` branch, and the bug-fixes should be actively backported to the
```

Comment:
This is confusing / not right. It is not that all PRs should be against `master` but that all *changes* should be against `master`.

Reply:
Is that even the case? Perhaps we will want to change something in an older version e.g. in a feature which has been removed in `master`.